### DISCUSSION TITLE
Clangd secure settings and automation

### DISCUSSION THREAD

**Discussion by tru** [clangd/clangd#1488] [General]
Hi,

Internally we are testing a new VSCode extension that integrates with our internal build system and clangd to get code indexing and distributed indexes. In our extension, you can select the currently active configuration. To get the indexing to match, we update clangd's settings (pointing it to another directory with another `compile_commands.json`) and restart it by sending `clangd.restart` command. But since we modify the settings, we will get the notification that the settings deviate from the default settings, which has caused some confusion from our internal beta testers.

While I understand this feature to be a good security practice, I wonder if there is a way to circumvent this somehow or start the plugin in an "insecure" mode.

My only other thought is to fork the plugin to remove this and distribute this fork internally, which I hope to avoid for obvious reasons.

Any feedback?

**Comment by HighCommander4** [comment]
There was a recent discussion about this in https://github.com/clangd/vscode-clangd/issues/318. It sounds like there is consensus to move away from this functionality in vscode-clangd in favour of vscode's "workspace trust" feature, it's just a matter of someone writing a patch to do it.

**Comment by tru** [comment]
I'll take a crack at this then.
